src/dali/plugin/input/rest/binance_us.py

```diff
@@ -0,0 +1,1329 @@
+_INTEREST_SIZE_LIMIT: int = 100
+_MINING_PAGE_LIMIT: int = 200
+_TRADE_RECORD_LIMIT: int = 1000
+_WITHDRAWAL_RECORD_LIMIT: int = 1000
+
+# Types of Binance Dividends
+_BNB_VAULT = "BNB Vault"
+_ETH_STAKING = "ETH 2.0 Staking"
+_FLEXIBLE = "Flexible"
+_FLEXIBLE_SAVINGS = "Flexible Savings"
+_LAUNCH_POOL = "Launchpool"
+_LOCKED = "Locked"
+_LOCKED_SAVINGS = "Locked Savings"
+_LOCKED_STAKING = "Locked Staking"
+_SOLO_AIRDROP = "SOLO airdrop"
+_GENERAL_STAKING = "STAKING"
+# More types added
+_CASH_VOUCHER = "Cash Voucher"
+_LAUNCHPAD = "launchpad"
+_SAVINGS_TRAIL_FUND = "Savings Trail Fund"
+
+_AIRDROP_LIST = [_SOLO_AIRDROP]
+_INTEREST_LIST = [_FLEXIBLE, _FLEXIBLE_SAVINGS, _LOCKED, _LOCKED_SAVINGS, _SAVINGS_TRAIL_FUND]
+_STAKING_LIST = [_ETH_STAKING, _LOCKED_STAKING, _BNB_VAULT, _LAUNCH_POOL, _GENERAL_STAKING, _LAUNCHPAD]
+_INCOME_LIST = [_CASH_VOUCHER]
+
+
+class InputPlugin(AbstractCcxtInputPlugin):
+    __EXCHANGE_NAME: str = "Binance.us"
+    __PLUGIN_NAME: str = "Binance.us_REST"
+    __DEFAULT_THREAD_COUNT: int = 1
+
+    def __init__(
+        self,
+        account_holder: str,
+        api_key: str,
+        api_secret: str,
+        native_fiat: str,
+        username: Optional[str] = None,
+        thread_count: Optional[int] = __DEFAULT_THREAD_COUNT,
+    ) -> None:
+        self.__api_key = api_key
+        self.__api_secret = api_secret
+        # We will have a default start time of July 13th, 2017 since Binance Exchange officially launched on July 14th Beijing Time.
+        super().__init__(account_holder, datetime(2017, 7, 13, 0, 0, 0, 0), native_fiat, thread_count)
+        self.__username = username
+
+        # We have to know what markets and algos are on Binance so that we can pull orders using the market
+        self.__algos: List[str] = []
+
+    def exchange_name(self) -> str:
+        return self.__EXCHANGE_NAME
+
+    def plugin_name(self) -> str:
+        return self.__PLUGIN_NAME
+
+    def _initialize_client(self) -> binanceus:
+        return binanceus(
+            {
+                "apiKey": self.__api_key,
+                "enableRateLimit": True,
+                "secret": self.__api_secret,
+            }
+        )
+
+    @staticmethod
+    def _to_trade(market_pair: str, base_amount: str, quote_amount: str) -> Trade:
+        assets = market_pair.split("/")
+        if assets[0] == "USD4":
+            assets[0] = "USD"
+        if assets[1] == "USD4":
+            assets[1] = "USD"
+        return Trade(
+            base_asset=assets[0],
+            quote_asset=assets[1],
+            base_info=f"{base_amount} {assets[0]}",
+            quote_info=f"{quote_amount} {assets[1]}",
+        )
+
+    @property
+    def _client(self) -> binanceus:
+        super_client: Exchange = super()._client
+        if not isinstance(super_client, binanceus):
+            raise RP2RuntimeError("Exchange is not instance of class binanceus.")
+        return super_client
+
+    def _get_algos(self) -> List[str]:
```

Comment:
This function is associated with mining so it can be deleted too if Binance.us doesn't deal with mining.